addons/mrp/wizard/mrp_product_produce.py

```diff
@@ -118,6 +120,37 @@ def do_produce(self):
             })
         return {'type': 'ir.actions.act_window_close'}
 
+    def _set_quantity_done(self, move, qty):
+        """
+        Set the given quantity as quantity done on the move through the move lines. The method is
+        able to handle move lines with a different UoM than the move (but honestly, this would be
+        looking for trouble...).
+        @param move: stock.move
+        @param qty: quantity in the UoM of move.product_uom
+        """
+        for ml in move.move_line_ids:
+            # Convert move line qty into move uom
+            ml_qty = ml.product_uom_qty
+            if ml.product_uom_id != move.product_uom:
+                ml_qty = ml.product_uom_id._compute_quantity(ml_qty, move.product_uom, round=False)
+
+            taken_qty = min(qty, ml_qty)
+            # Convert taken qty into move line uom
+            if ml.product_uom_id != move.product_uom:
+                taken_qty = move.product_uom._compute_quantity(ml_qty, ml.product_uom_id, round=False)
+
+            # Assign qty_done and explicitly round to make sure there is no inconsistency between
+            # ml.qty_done and qty.
+            taken_qty = float_round(taken_qty, precision_rounding=ml.product_uom_id.rounding)
+            ml.qty_done = taken_qty
+            if ml.product_uom_id != move.product_uom:
+                taken_qty = ml.product_uom_id._compute_quantity(ml_qty, move.product_uom, round=False)
+            qty -= taken_qty
+
+            if float_compare(qty, 0.0,  precision_rounding=move.product_uom.rounding) <= 0:
+                break
+        return True
```

Comment:
It is possible that you have to consume more quantities than reserved. In this case you should put the remaining quantities in a new move line or an existing one (if possible in the MO locations in order to avoid negative quantity in a sublocation.)